dissect/volume/disk/partition.py

```diff
@@ -224,16 +224,14 @@ def __init__(
     def __repr__(self) -> str:
         type_str = self.type_str
 
-        if isinstance(self.type, int):
-            type = hex(self.type)
-            type_key = self.type
-        elif isinstance(self.type, bytes):
-            type = UUID(bytes_le=self.type)
-            type_key = type
-        elif isinstance(self.type, str):
-            type_str = self.type
+        if type_str is None:
+            if isinstance(self.type, int):
+                type_str = f"{hex(self.type)} ({PARTITION_TYPES.get(self.type, 'Unknown')})"
+            elif isinstance(self.type, UUID):
+                type_str = f"{self.type} ({PARTITION_TYPES.get(self.type, 'Unknown')})"
+            else:
+                type_str = self.type
 
-        type_str = type_str or f"{type} ({PARTITION_TYPES.get(type_key, 'Unknown')})"
         return (
             f"<Partition number={self.number} offset=0x{self.offset:x} "
             f"size=0x{self.size:x} type={type_str} name={self.name!r}>"
```

Comment:
```suggestion
            f"size=0x{self.size:x} type={type_str!r} name={self.name!r}>"
```
As `type_str` can sometimes contain spaces, maybe using the repr makes it a bit more clear what belongs to `type=`